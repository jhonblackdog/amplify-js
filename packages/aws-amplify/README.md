# AWS Amplify Package - aws-amplify

AWS Amplify core library
