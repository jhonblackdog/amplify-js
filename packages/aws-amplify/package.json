{
  "name": "aws-amplify",
  "version": "1.1.40",
  "description": "AWS Amplify is a JavaScript library for Frontend and mobile developers building cloud-enabled applications.",
  "main": "./index.js",
  "module": "./lib-esm/index.js",
  "typings": "./lib-esm/index.d.ts",
  "react-native": {
    "./index": "./lib-esm/index.js"
  },
  "scripts": {
    "test": "jest --passWithNoTests --coverage --maxWorkers 2",
    "build-with-test": "npm run clean && npm test && tsc && webpack -p",
    "build:cjs": "node ./build es5 && webpack && webpack --config ./webpack.config.dev.js",
    "build:esm": "node ./build es6",
    "build": "npm run clean && npm run build:esm && npm run build:cjs",
    "clean": "rimraf lib-esm lib dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'",
    "generate-docs-local": "typedoc --out docs src",
    "generate-docs-root": "typedoc --out ../../docs src"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "dependencies": {
    "@aws-amplify/analytics": "^1.2.25",
    "@aws-amplify/api": "^1.1.3",
    "@aws-amplify/auth": "^1.3.3",
    "@aws-amplify/cache": "^1.0.34",
    "@aws-amplify/core": "^1.1.2",
    "@aws-amplify/interactions": "^1.0.37",
    "@aws-amplify/predictions": "^1.0.5",
    "@aws-amplify/pubsub": "^1.1.2",
    "@aws-amplify/storage": "^1.1.2",
    "@aws-amplify/ui": "^1.0.25",
    "@aws-amplify/xr": "^0.1.24"
  }
}
