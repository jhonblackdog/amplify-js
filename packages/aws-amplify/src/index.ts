/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

export {
    Analytics,
    AnalyticsClass,
    AnalyticsProvider,
    AWSPinpointProvider, 
    AWSKinesisProvider } from '@aws-amplify/analytics';
export { Auth, AuthClass } from '@aws-amplify/auth';
export { Storage, StorageClass } from '@aws-amplify/storage';
export { API, APIClass, graphqlOperation } from '@aws-amplify/api';
export { PubSub, PubSubClass } from '@aws-amplify/pubsub';
export { default as Cache } from '@aws-amplify/cache';
export { Interactions, InteractionsClass } from '@aws-amplify/interactions';
export  * from '@aws-amplify/ui';
export { XR, XRClass } from '@aws-amplify/xr';
export { Predictions, AmazonAIPredictionsProvider } from '@aws-amplify/predictions';
export { 
    Amplify, 
    ConsoleLogger as Logger,
    Hub,
    JS,
    ClientDevice,
    Signer,
    I18n,
    ServiceWorker
} from '@aws-amplify/core';

import { 
    Amplify, 
    I18n, // Deprecated: Remove on major version bump
    Logger, // Deprecated: Remove on major version bump
    ServiceWorker
} from '@aws-amplify/core';

/* Deprecated: Remove on major version bump */
import Analytics from '@aws-amplify/analytics';
import Auth from '@aws-amplify/auth';
import Storage from '@aws-amplify/storage';
import API from '@aws-amplify/api';
import PubSub from '@aws-amplify/pubsub';
import Cache from '@aws-amplify/cache';
import Interactions from '@aws-amplify/interactions';
import * as UI from '@aws-amplify/ui';
import XR from '@aws-amplify/xr';
import Predictions from '@aws-amplify/predictions';
/* */

/* Deprecated: Remove on major version bump */
Amplify.Auth = Auth;
Amplify.Analytics = Analytics;
Amplify.API = API;
Amplify.Storage = Storage;
Amplify.I18n = I18n;
Amplify.Cache = Cache;
Amplify.PubSub = PubSub;
Amplify.Logger = Logger;
Amplify.ServiceWorker = ServiceWorker;
Amplify.Interactions = Interactions;
Amplify.UI = UI;
Amplify.XR = XR;
Amplify.Predictions = Predictions;
/* */

export default Amplify;
