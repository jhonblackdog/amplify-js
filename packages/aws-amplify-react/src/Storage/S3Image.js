/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';

import { Storage, Logger } from 'aws-amplify';

import AmplifyTheme from '../AmplifyTheme';

const logger = new Logger('Storage.S3Image');

export default class S3Image extends Component {
    constructor(props) {
        super(props);

        this.handleOnLoad = this.handleOnLoad.bind(this);

        this.state = { src: null };
    }

    getImageSource() {
        const { path, level } = this.props;
        Storage.get(path, { level: level? level : 'public' })
            .then(url => {
                this.setState({
                    src: url
                });
            })
            .catch(err => logger.warn(err));
    }

    load() {
        const { path, body, contentType, level } = this.props;
        if (!path) {
            logger.debug('empty path');
            return ;
        }

        const that = this;
        logger.debug('loading ' + path + '...');
        if (body) {
            const type = contentType || 'binary/octet-stream';
            const ret = Storage.put(path, body, type, { level: level? level : 'public' });
            ret.then(data => {
                logger.debug(data);
                that.getImageSource();
            })
            .catch(err => logger.warn(err));
        } else {
            that.getImageSource();
        }
    }

    handleOnLoad(evt) {
        const { onReady } = this.props;
        if (onReady) { onReady(this.state.src); }
    }

    componentDidMount() {
        this.load();
    }

    componentDidUpdate(prevProps) {
        if (prevProps.path !== this.props.path || prevProps.body !== this.props.body) {
            this.load();
        }
    }

    render() {
        const { src } = this.state;
        if (!src) { return null; }

        const theme = this.props.theme || AmplifyTheme;
        const { hidden, style } = this.props;
        const imgStyle = hidden? AmplifyTheme.hidden
                            : Object.assign({}, theme.photo, style);

        return <img style={imgStyle} src={src} onLoad={this.handleOnLoad} />
    }
}
