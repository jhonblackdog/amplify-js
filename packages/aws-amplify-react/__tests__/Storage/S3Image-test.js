import S3Image from '../../src/Storage/S3Image';
import React from 'react';
import { Logger, Storage } from 'aws-amplify';

describe('S3Image', () => {
    test('render correctly if has srcs', () => {
        const wrapper = shallow(<S3Image/>);

        wrapper.setState({src: 'src'});
        expect(wrapper).toMatchSnapshot();
    });

    test('render nothing if no src', () => {
        const wrapper = shallow(<S3Image/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('mount S3 image with no path', () => {
        const spyon = jest.spyOn(Logger.prototype, 'debug');
        const wrapper = mount(<S3Image/>);

        expect(spyon).toBeCalled();
    });

    test('mount S3 image with path', async () => {
        const spyon = jest.spyOn(Storage, 'put').mockImplementationOnce(() => {
            return new Promise((res, rej) => {
                res('data');
            });
        });
        const wrapper = await mount(<S3Image path={'path'} body={'something'} />);

        expect(spyon).toBeCalled();

        spyon.mockClear();
    });

    test('mount S3 image with path and error when putting', async () => {
        const spyon = jest.spyOn(Storage, 'put').mockImplementationOnce(() => {
            return new Promise((res, rej) => {
                rej('err');
            });
        });
        const wrapper = await mount(<S3Image path={'path'} body={'something'} />);

        expect(spyon).toBeCalled();

        spyon.mockClear();
    });

    test('mount S3 image with path and error when putting', async () => {
        const spyon = jest.spyOn(Storage, 'get').mockImplementationOnce(() => {
            return new Promise((res, rej) => {
                res('data');
            });
        });
        const wrapper = await mount(<S3Image path={'path'} body={null} />);

        expect(spyon).toBeCalled();

        spyon.mockClear();
    });

});
