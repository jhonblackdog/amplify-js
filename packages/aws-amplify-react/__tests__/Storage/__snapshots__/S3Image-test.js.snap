// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S3Image render correctly if has srcs 1`] = `
<img
  onLoad={[Function]}
  src="src"
  style={
    Object {
      "display": "inline-block",
      "margin": "0.2%",
      "verticalAlign": "top",
      "width": "32.5%",
    }
  }
/>
`;

exports[`S3Image render nothing if no src 1`] = `""`;
